indra/newview/llchathistory.cpp

```diff
@@ -1227,26 +1225,17 @@ void LLChatHistory::appendMessage(const LLChat& chat, const LLSD &args, const LL
 		mUnreadChatSources.insert(chat.mFromName);
 		mMoreChatPanel->setVisible(TRUE);
 		std::string chatters;
-		for (unread_chat_source_t::iterator it = mUnreadChatSources.begin();
-			it != mUnreadChatSources.end();)
+		for (std::string source : mUnreadChatSources)
```

Comment:
mac Os should have an issue with this one being a referenece (it has no issues doing the same for mNameList), but if does, I suggest reverting to 'iterator' variant.